<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <meta name="description" content="">
        <meta name="author" content="">
        <title>Lecture 2</title>
        <!-- Bootstrap core CSS -->
        <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
        <link rel="manifest" href="/site.webmanifest">
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5">
        <meta name="msapplication-TileColor" content="#da532c">
        <meta name="theme-color" content="#ffffff">
        <!-- Bootstrap core CSS -->
        <link href="bootstrap/css/bootstrap.css" rel="stylesheet">
        <!-- Custom styles for this template -->
        <link href="https://fonts.googleapis.com/css?family=Playfair+Display:700,900" rel="stylesheet">
        <link href="styles.css" rel="stylesheet">
        <link href="prism.css" rel="stylesheet">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Condensed">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Mono">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Slab">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Dosis">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Quicksand">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto">
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css">
    </head>
    <body>
        <div class="container">
            <header class="blog-header py-3 mb-4 pt-3 pb-3">
                <h1 style="font-family: Roboto Condensed; font-weight: bold;" class="display-5 mb-0 text-center">Lecture 2</h1>
            </header>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse1" aria-expanded="false" aria-controls="collapse1">Copy the files for today's class from <code>/home/hgen_share/lec2</code></a></h5>
                </div>
                <div id="collapse1" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Set up directory and copy over files</code></code></p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse2" aria-expanded="false" aria-controls="collapse2">Compare <code>HG002.sorted.bam</code> with what you generated last week</a></h5>
                </div>
                <div id="collapse2" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p> <b> Remember to load modules </b> </p>
                        <hr/>
                        <p>Index the <code>bam</code> file</p>
                        <hr/>
                        <p>Run <kbd>samtools flagstat</kbd> on both files</p>
                        <hr/>
                        What are some of the similarities and differences?
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse3" aria-expanded="false" aria-controls="collapse3">Visualize <code>HG002.sorted.bam</code> in IGV</a></h5>
                </div>
                <div id="collapse3" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Download the <code>bam</code> file to your local computer as previously shown using <code>scp</code></code> </code></p>
                        <hr/>
                        <p>Check what assembly it was aligned to</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">samtools view -H HG002.sorted.bam | tail -n 2</code></pre>
                        <hr/>
                        Open it in IGV with the appropriate reference
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse6" aria-expanded="false" aria-controls="collapse6">Re-align <code>HG002.sorted.bam</code> around susceptible regions with GATK</a></h5>
                </div>
                <div id="collapse6" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                      <p>Assign paths to variables for simplicity</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">REF=/home/hgen_share/bwa
VCF=/home/hgen_share/vcf/b37
PICARD_JAR=/home/hgen_share/utils/picard.jar
GATK_JAR=/home/hgen_share/utils/GenomeAnalysisTK-3.8.1.jar
export PATH="/home/hgen_share/utils/gatk-4.1.9.0:$PATH"
module load StdEnv/2020 samtools/1.16
module load nixpkgs/16.09 java/1.8.0_192</code></pre>
                        <p>Find regions with indels and dense in SNPs with <kbd>RealignerTargetCreator</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">java -Xmx4G -jar $GATK_JAR \
  -T RealignerTargetCreator \
  -R $REF/b37.fa \
  -L 20:43200000-43300000 \
  --known $VCF/dbSNP.vcf \
  --known $VCF/Mills.indels.vcf \
  -o realign.intervals \
  -I HG002.sorted.bam</code></pre>
                        <hr/>
                        <p>Re-align around the identified regions with <kbd>IndelRealigner</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">java -Xmx4G -jar $GATK_JAR \
  -T IndelRealigner \
  -R $REF/b37.fa \
  -known $VCF/dbSNP.vcf \
  -known $VCF/Mills.indels.vcf \
  -targetIntervals realign.intervals \
  -o HG002.realigned.sorted.bam \
  -I HG002.sorted.bam</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse7" aria-expanded="false" aria-controls="collapse7">Visualize re-aligned reads in IGV</a></h5>
                </div>
                <div id="collapse7" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Retrieve re-aligned reads based with "OC" tag ("original CIGAR", implying that a new CIGAR string was generated)</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">samtools view HG002.realigned.sorted.bam | grep 'OC' | cut -f1 | sort &gt oc.txt</code></pre>
                        <hr/>
                        <p>Sort reads by name using <kbd>picard SortSam</kbd> to allow efficient searching</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">java -Xmx4G -jar $PICARD_JAR SortSam \
  -I HG002.realigned.sorted.bam \
  -O HG002.realigned.qs.bam \
  -SO queryname
&nbsp
java -Xmx2g -jar $PICARD_JAR SortSam \
  -I HG002.sorted.bam \
  -O HG002.qs.bam \
  -SO queryname</code></pre>
                        <hr/>
                        <p>Subset reads with name matching those found in <code>oc.txt</code> with <kbd>picard FilterSamReads</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">java -Xmx4G -jar $PICARD_JAR FilterSamReads \
  -I HG002.realigned.qs.bam \
  -O HG002.new.bam \
  --FILTER includeReadList \
  -RLF oc.txt \
  -SO coordinate \
  -CREATE_INDEX true
&nbsp
java -Xmx2g -jar $PICARD_JAR FilterSamReads \
  -I HG002.qs.bam \
  -O HG002.old.bam \
  --FILTER includeReadList \
  -RLF oc.txt \
  -SO coordinate \
  -CREATE_INDEX true</code></pre>
                        <hr/>
                        <p>Download the new and old <code>bam</code> files. What else must you download with your <code>bam</code> file for visualization?</code></p>
                        <hr/>
                        Go to chr20:43,240,000-43,260,000 to see an obvious example.
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse8" aria-expanded="false" aria-controls="collapse8">Mark PCR duplicates with Picard</a></h5>
                </div>
                <div id="collapse8" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Mark PCR duplicate reads with <kbd>MarkDuplicates</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">java -Xmx4G -jar ${PICARD_JAR} MarkDuplicates \
  -REMOVE_DUPLICATES false \
  -CREATE_INDEX true \
  -I HG002.realigned.sorted.bam \
  -O HG002.sorted.dup.bam \
  -M HG002.sorted.dup.metrics</code></pre>
                        <hr/>
                        <p>Check the results</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">grep 'METRICS CLASS' -A 2 HG002.sorted.dup.metrics | column -t | less -S</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse9" aria-expanded="false" aria-controls="collapse9">Apply base quality score recalibration with GATK</a></h5>
                </div>
                <div id="collapse9" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Establish a set of covariates for correction based on context and known SNPs with <kbd>BaseRecalibrator</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">gatk --java-options "-Xmx4G" \
  BaseRecalibrator \
  -R $REF/b37.fa \
  -known-sites $VCF/dbSNP.vcf \
  -known-sites $VCF/Mills.indels.vcf \
  -L 20:43200000-43300000 \
  -O HG002.sorted.dup.recalibration_report.grp \
  -I HG002.sorted.dup.bam</code></pre>
                        <hr/>
                        <p>Recalibrate using the metrics calculated above with <kbd>PrintReads</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">gatk --java-options "-Xmx4G" \
  ApplyBQSR \
  -R $REF/b37.fa \
  --bqsr-recal-file HG002.sorted.dup.recalibration_report.grp \
  -O HG002.sorted.dup.recal.bam \
  -I HG002.sorted.dup.bam
   </code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse10" aria-expanded="false" aria-controls="collapse10">Call variants with GATK and visualize results in IGV</a></h5>
                </div>
                <div id="collapse10" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Perform variant calling with <kbd>HaplotypeCaller</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">gatk --java-options "-Xmx4G" \
  HaplotypeCaller \
  -R $REF/b37.fa \
  -I HG002.sorted.dup.recal.bam \
  -ERC GVCF \
  -O HG002.g.vcf \
  -L 20:43200000-43300000 \
  --native-pair-hmm-threads 1</code></pre>
                        <hr/>
                        <p>Download the <code>vcf</code> and recalibrated <code>bam</code> file to your local computer</p>
                        <p> More information, including parameter details, for GATK <code>HaplotypeCaller</code> found <a class="headref" href="https://gatk.broadinstitute.org/hc/en-us/articles/360037225632-HaplotypeCaller">here</a> </p>
                    </div>
                </div>
            </div>
        <!-- Bootstrap core JavaScript
    ================================================== -->
        <!-- Placed at the end of the document so the pages load faster -->
        <script src="assets/js/jquery.min.js"></script>
        <script src="assets/js/popper.js"></script>
        <script src="bootstrap/js/bootstrap.min.js"></script>
        <script src="assets/js/prism.js"></script>
        <script src="assets/js/extra.js"></script>
        <a id="back-to-top" href="#" class="btn btn-primary btn-lg back-to-top" role="button" data-toggle="tooltip" data-placement="left"><span class="fas fa-angle-up"></span></a>
    </body>
</html>
