<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <meta name="description" content="">
        <meta name="author" content="">
        <title>Lecture 3</title>
        <!-- Bootstrap core CSS -->
        <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
        <link rel="manifest" href="/site.webmanifest">
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5">
        <meta name="msapplication-TileColor" content="#da532c">
        <meta name="theme-color" content="#ffffff">
        <!-- Bootstrap core CSS -->
        <link href="bootstrap/css/bootstrap.css" rel="stylesheet">
        <!-- Custom styles for this template -->
        <link href="https://fonts.googleapis.com/css?family=Playfair+Display:700,900" rel="stylesheet">
        <link href="styles.css" rel="stylesheet">
        <link href="prism.css" rel="stylesheet">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Condensed">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Mono">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Slab">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Dosis">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Quicksand">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto">
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css">
    </head>
    <body>
        <div class="container">
            <header class="blog-header py-3 mb-4 pt-3 pb-3">
                <h1 style="font-family: Roboto Condensed; font-weight: bold;" class="display-5 mb-0 text-center">Lecture 3</h1>
            </header>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse1" aria-expanded="false" aria-controls="collapse1">Take a look at <code>HG002.g.vcf</code> from last week</a></h5>
                </div>
                <div id="collapse1" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>View the file with <code>less</code></p>
                        <p>Create a new directory and move the <code>vcf</code> file and its index to the new directory using <code>mv</code></p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse3" aria-expanded="false" aria-controls="collapse3">Filter variants using GATK recommanded cut-offs</a></h5>
                </div>
                <div id="collapse3" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Set up variables</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">REF=/home/hgen_share/bwa
VCF=/home/hgen_share/vcf/b37
PICARD_JAR=/home/hgen_share/utils/picard.jar
GATK_JAR=/home/hgen_share/utils/GenomeAnalysisTK-3.8.1.jar
export PATH="/home/hgen_share/utils/gatk-4.1.9.0:$PATH"
module load StdEnv/2020 samtools/1.16
module load nixpkgs/16.09 java/1.8.0_192</code></pre>
                        <hr/>
                        <p>Replace the hashtag (#) with the recommended cut-offs found <a class="headref" href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890471-Hard-filtering-germline-short-variants">here</a></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">gatk --java-options "-Xmx4G" \
    VariantFiltration \
    -V HG002.g.vcf \
    -filter "QD &lt #" --filter-name "QD" \
    -filter "FS &gt #" --filter-name "FS" \
    -filter "SOR &gt #" --filter-name "SOR" \
    -filter "MQ &lt #" --filter-name "MQ" \
    -filter "MQRankSum &lt #" --filter-name "MQRankSum" \
    -filter "ReadPosRankSum &lt #" --filter-name "ReadPosRankSum" \
    -O HG002.filt.g.vcf 2> >(grep -v undefined)</code></pre>
                        <hr/>
                        <p>How many variants appear ok?</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">grep -v '^#' HG002.filt.g.vcf | cut -f7 | grep 'PASS' | wc -l</code></pre>
                        <hr/>
                        <p>How many total variants are called? Hint: use <code>grep</code></p>
                        <hr/>
                        <p>Set up variables for <a class="headref" href="https://github.com/google/deepvariant">DeepVariant</a></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">module load StdEnv/2020 apptainer/1.1.3
mkdir -p "${PWD}/data"
mkdir -p "${PWD}/output"
INPUT_DIR="${PWD}/data"
OUTPUT_DIR="${PWD}/output"
TMPDIR=.
cp /home/hgen_share/lec3/* ${INPUT_DIR}</code></pre>
                        <hr/>
                        <p>Run <a class="headref" href="https://github.com/google/deepvariant">DeepVariant</a> on the same sample from last week. <b>Note: this may take a few minutes</b></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">singularity run -B $PWD,/usr/lib/locale/ \
   /home/hgen_share/deepvariant/deepvariant_1.4.0.sif \
   /opt/deepvariant/bin/run_deepvariant \
   --model_type=WGS \
   --ref="${INPUT_DIR}"/b37.chr20.fa \
   --reads="${INPUT_DIR}"/HG002.sorted.bam \
   --regions "20:43200000-43300000" \
   --output_vcf="${OUTPUT_DIR}"/HG002.dv.vcf.gz \
   --output_gvcf="${OUTPUT_DIR}"/HG002.dv.g.vcf.gz \
   --num_shards=1</code></pre>
                        <hr/>
                        <p>Unzip the <code>g.vcf</code> found in the <code>output</code> directory using <code>gunzip</code> then filter it using similar parameters as above. Compare the number of variants that pass. <b>Note: you may need to set-up your variables for GATK again</b></p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse4" aria-expanded="false" aria-controls="collapse4">(Extra) Compare hard filtering with VQSR</a></h5>
                </div>
                <div id="collapse4" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                      <p>Copy whole genome variant calls</p>
                      <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">cp /home/hgen_share/lec3_part2 ./</code></pre>
                      <hr/>
                        <p>Since SNPs and indels need to be trained separately, start with SNPs</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">gatk --java-options "-Xmx4G" \
  VariantRecalibrator \
   -R ${REF}/b37.fa \
   -V raw.vcf \
   --resource:hapmap,known=false,training=true,truth=true,prior=15.0 $VCF/b37/hapmap.vcf \
   --resource:omni,known=false,training=true,truth=false,prior=12.0 $VCF/b37/omni.vcf \
   --resource:1000G,known=false,training=true,truth=false,prior=10.0 $VCF/b37/1KG.snps.vcf \
   --resource:dbsnp,known=true,training=false,truth=false,prior=2.0 $VCF/b37/dbSNP.vcf \
   -an DP -an QD -an FS -an SOR -an MQ -an MQRankSum -an ReadPosRankSum \
   -mode SNP \
   -O recal_SNP.recal \
   --tranches-file recal_SNP.tranches \
   --rscript-file recal_SNP.plots.R</code></pre>
                        <hr/>
                        <p>Apply VQSR to SNPs</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">gatk --java-options "-Xmx4G" \
  ApplyVQSR \
  -R ${REF}/b37.fa \
  -V raw.vcf \
  -O raw.snp.vcf \
  --truth-sensitivity-filter-level 99.5 \
  --tranches-file recal_SNP.tranches \
  --recal-file recal_SNP.recal \
  -mode SNP</code></pre>
                        <hr/>
                        <p>Now repeat for indels</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">gatk --java-options "-Xmx4G" \
  VariantRecalibrator \
  -R ${REF}/b37.fa \
  -V raw.snp.vcf \
  --resource:mills,known=false,training=true,truth=true,prior=12.0 $VCF/b37/Mills.indels.vcf \
  --resource:dbsnp,known=true,training=false,truth=false,prior=2.0 $VCF/b37/dbSNP.vcf \
  --max-gaussians 4 \
  -an QD -an DP -an FS -an SOR -an MQRankSum -an ReadPosRankSum \
  -mode INDEL \
  -O recal_INDEL.recal \
  --tranches-file recal_INDEL.tranches \
  --rscript-file recal_INDEL.plots.R
&nbsp
gatk --java-options "-Xmx4G" \
  ApplyVQSR \
  -R ${REF}/b37.fa \
  -V raw.snp.vcf \
  -O recal.vcf \
  --truth-sensitivity-filter-level 99.0 \
  --tranches-file recal_INDEL.tranches \
  --recal-file recal_INDEL.recal \
  -mode INDEL</code></pre>
                        <hr/>
                        <p>How many variants in the same region passed?</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">gatk SelectVariants \
  -R ${REF}/b37.fa \
  -V recal.vcf \
  -O recal.sub.vcf \
  -L 20:43200000-43300000
&nbsp
grep -v '^#' recal.sub.vcf | cut -f7 | grep 'PASS' | wc -l</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse6" aria-expanded="false" aria-controls="collapse6">(Extra) Generate a VCF file with only variants that passed</a></h5>
                </div>
                <div id="collapse6" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Set up <kbd>bcftools</kbd>to filter PASS variants</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">module load StdEnv/2020 gcc/9.3.0 bcftools/1.16</code></pre>
                        <hr/>
                        <p>Run <kbd>bcftools</kbd>. The input file can be the VCF file produced from either variant caller</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">bcftools view -f 'PASS,.' -O v -o output_filename input_file </code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse6" aria-expanded="false" aria-controls="collapse6">(Extra) Use <code>VariantsToTable</code>to get a tsv file of variants</a></h5>
                </div>
                <div id="collapse6" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p><code>VariantsToTable</code>, by default, only extracts variants that passed filtering. Use the --show-filtered parameter to show all variants.</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">REF=/home/hgen_share/bwa
gatk VariantsToTable \
    -R $REF/b37.fa \
    -V input_vcf \
    -F CHROM -F POS -F FILTER -F TYPE -GF AD -GF DP \
    --show-filtered \
    -O output_filename.tsv</code></pre>
    <p>You may need to reload <code>modules</code>for GATK again</p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse5" aria-expanded="false" aria-controls="collapse5">Add functional annotations with SnpEff</a></h5>
                </div>
                <div id="collapse5" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Copy whole genome variant calls </p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code>cp /home/hgen_share/lec3_part2/* ./</code></pre>
                        <hr/>
                        <p>Use <kbd>SnpEff</kbd> with mostly default parameters</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">UTILS=/home/partage/utils
SNPEFF_JAR=/home/hgen_share/utils/snpEff/snpEff.jar
java -Xmx4G -jar $SNPEFF_JAR \
  -c /home/hgen_share/utils/snpEff/snpEff.config \
  -i vcf -o vcf -v hg19 \
  ex.vcf &gt ex.snpeff.vcf</code></pre>
                        <hr/>
                        <p>Download <code>.html</code> results file to your local computer using <code>scp</code></p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse7" aria-expanded="false" aria-controls="collapse7">Annotate with existing resources using GEMINI</a></h5>
                </div>
                <div id="collapse7" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                      <p>Add the directory containing <kbd>gemini</kbd></p>
                      <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">export PATH=/home/hgen_share/utils/gemini/anaconda/bin:$PATH</code></pre>
                      <hr/>
                        <p>Generate annotated database with <kbd>gemini</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">module load mugqic/gemini/0.20.1
gemini load --cores 4 --tempdir $PWD -t snpEff -v ex.snpeff.vcf ex.snpeff.db</code></pre>
                        <hr/>
                        <p>Dump database to plain-text csv using <kbd>sqlite</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">db=ex.snpeff.db
t=($(sqlite3 $db ".tables"))
for i in "${t[@]}"; do
	sqlite3 $db&lt&lt- EXIT_HERE
	.mode csv
	.headers on
	.output $i.csv
	SELECT * FROM $i;
	.exit
	EXIT_HERE
  echo "$i.csv generated"
done</code></pre>
                        <hr/>
                        <p>Download <code>csv</code> file</p>
                        <hr/>
                        <p> For more details regarding columns: <a class="headref" href="https://gemini.readthedocs.io/en/latest/content/database_schema.html">GEMINI database</a> </p>
                    </div>
                </div>
            </div>
        <!-- Bootstrap core JavaScript
    ================================================== -->
        <!-- Placed at the end of the document so the pages load faster -->
        <script src="assets/js/jquery.min.js"></script>
        <script src="assets/js/popper.js"></script>
        <script src="bootstrap/js/bootstrap.min.js"></script>
        <script src="assets/js/prism.js"></script>
        <script src="assets/js/extra.js"></script>
        <a id="back-to-top" href="#" class="btn btn-primary btn-lg back-to-top" role="button" data-toggle="tooltip" data-placement="left"><span class="fas fa-angle-up"></span></a>
    </body>
</html>
