<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <meta name="description" content="">
        <meta name="author" content="">
        <title>Lecture 4</title>
        <!-- Bootstrap core CSS -->
        <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
        <link rel="manifest" href="/site.webmanifest">
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5">
        <meta name="msapplication-TileColor" content="#da532c">
        <meta name="theme-color" content="#ffffff">
        <!-- Bootstrap core CSS -->
        <link href="bootstrap/css/bootstrap.css" rel="stylesheet">
        <!-- Custom styles for this template -->
        <link href="https://fonts.googleapis.com/css?family=Playfair+Display:700,900" rel="stylesheet">
        <link href="styles.css" rel="stylesheet">
        <link href="prism.css" rel="stylesheet">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Condensed">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Mono">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Slab">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Dosis">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Quicksand">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto">
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css">
    </head>
    <body>
        <div class="container">
            <header class="blog-header py-3 mb-4 pt-3 pb-3">
                <h1 style="font-family: Roboto Condensed; font-weight: bold;" class="display-5 mb-0 text-center">Lecture 4</h1>
            </header>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse1" aria-expanded="false" aria-controls="collapse1">Copy files for part 1 from <code>/home/hgen_share/lec4_pt1</code></a></h5>
                </div>
                <div id="collapse1" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Set up directory and copy over files</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">module load StdEnv/2020 gcc/9.3.0 openmpi/4.0.3 hdf5/1.10.6 minimap2/2.17 fastqc/0.11.9 samtools/1.11 mugqic/qualimap/2.2.1 mafft/7.471 emboss/6.6.0 fasttree/2.1.11 mugqic/pycoQC/2.5.2
export PATH=/home/hgen_share/ont-guppy-cpu/bin:$PATH</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse2" aria-expanded="false" aria-controls="collapse2">Take a look at <code>ex.fast5</code></a></h5>
                </div>
                <div id="collapse2" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>View the file with <kbd>h5ls</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">h5ls ex.fast5 | less</code></pre>
                        <hr/>
                        <p>Show structure of one read using <kbd>h5ls</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">h5ls -r ex.fast5/read_00017b75-18fc-40ae-8a21-a5b3e49dc753</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse3" aria-expanded="false" aria-controls="collapse3">Basecall using Guppy</a></h5>
                </div>
                <div id="collapse3" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>This will take a while with just 1 CPU thread, and since the output is provided already, you can terminate it prematurely with <kbd>Ctrl-C</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">guppy_basecaller \
  --compress_fastq \
  -i ex.fast5 \
  -s basecall/ \
  --cpu_threads_per_caller 1 \
  --num_callers 1 \
  -c ${conf}.cfg</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse4" aria-expanded="false" aria-controls="collapse4">(Extra) Assess quality with pycoQC & FastQC</a></h5>
                </div>
                <div id="collapse4" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                      <p>Run <kbd>pycoQC</kbd> with the provided summary</p>
                      <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">pycoQC -f sequencing_summary.txt -o pyco.html</code></pre>
                      <hr/>
                        <p>Run <kbd>FastQC</kbd> with the provided sequences</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">fastqc ex.fq</code></pre>
                        <hr/>
                        <p>Download pycoQC <code>html</code> report</p>
                        <hr/>
                        <p>Compare with another report <a href="https://a-slide.github.io/pycoQC/pycoQC/results/Guppy-2.3_basecall-1D_alignment-DNA.html" target="_blank">here</a></p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse5" aria-expanded="false" aria-controls="collapse5">(Extra) Rectify quality issues by trimming</a></h5>
                </div>
                <div id="collapse5" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Trim adapters with <kbd>Porechop</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">porechop -i ex.fq -t 1 -v 2 -o chopped.fq</code></pre>
                        <hr/>
                        <p>Is it better now? Run fastQC on trimmed <code>fastq</code><p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse6" aria-expanded="false" aria-controls="collapse6">(Extra) Assess error profile after alignment</a></h5>
                </div>
                <div id="collapse6" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Align trimmed reads using <kbd>minimap2</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">minimap2 -t 1 -x map-ont -a wuhCor1.fa chopped.fq | samtools sort -o ex.bam
samtools index ex.bam</code></pre>
                        <hr/>
                        <p>Summarize alignment results with <kbd>Qualimap</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">qualimap bamqc -bam ex.bam --java-mem-size=1G -nw 5000 -nt 1 -c -outdir qualimap/</code></pre>
                        <hr/>
                        <p>Download results to your local computer</p>
                        <hr/>
                        <p>Compare with the <a href="https://rawgit.com/kokonech/kokonech.github.io/master/qualimap/HG00096.chr20_bamqc/qualimapReport.html" target="_blank">report</a> from a typical Illumina run</p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse7" aria-expanded="false" aria-controls="collapse7">Hybrid assembly with Unicycler</a></h5>
                </div>
                <div id="collapse7" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                      <p>Set up variables and copy over files for part 2</p>
                      <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">export PATH=/home/hgen_share/utils/SPAdes-3.15.5-Linux/bin:$PATH
export PATH=/home/hgen_share/utils/ncbi-blast-2.13.0+/bin:$PATH
export PATH=/home/hgen_share/utils/racon/build/bin:$PATH
cp /home/hgen_share/lec4_pt2/* ./</code></pre>
                      <hr/>
                      <p>Run <code>Unicycler</code> using boths short- and long-reads. For hybrid assemblies, the direct long-read bridging step of the pipeline can take a while to complete. Thus, the output is provided for you. You can terminate it prematurely with <code>Ctrl-C</code></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">unicycler -1 Illumina_F.fq -2 Illumina_R.fq -l ONT.fq -t 20 --min_fasta_length 10000 --linear_seqs 1 -o unicycler</code></pre>
                        <hr/>
                        <p>The output can be found in <kbd>/home/hgen_share/lec4_pt2/unicycler</kbd></p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse8" aria-expanded="false" aria-controls="collapse8">Evaluate the assembled genome's quality</a></h5>
                </div>
                <div id="collapse8" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Use <kbd>QUAST</kbd> to create a summary report</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">module load mugqic/Quast/5.0.2
quast.py -t 1 -o quast -R wuhCor1.fa -g wuhCor1.gff3 unicycler/assembly.fasta</code></pre>
                        <hr/>
                        <p>Compare with a <a href="http://cab.cc.spbu.ru/quast/paper/h.sapiens_chr14/" target="_blank">report</a> on eukaryotic assemblies</p>
                        <hr/>
                        <p><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="_blank">BLAST</a> the top 2 contigs (<code>contig1.fa</code> and <code>contig2.fa</code>) found in the <code>unicycler</code> output directory</p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse11" aria-expanded="false" aria-controls="collapse11">Place newly assembled viral genomes in a global context</a></h5>
                </div>
                <div id="collapse11" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Set up variables for nextstrain environment</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">export PATH="/home/hgen_share/Anaconda/bin:$PATH"
export PATH="/home/hgen_share/Anaconda/envs/nextstrain_V2/bin:$PATH"</code></pre>
                        <hr/>
                        <p>Copy the tutorial dataset for SARS-Cov2 provided by nextstrain</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">cp -r /home/hgen_share/ncov ./</code></pre>
                        <hr/>
                        <p>Take a look at the <kbd>/data</kbd> sub-directory. Inspect the data files with <code>less</code> to get an impression of what data needs to be provided for nextstrain. Unzip the fasta file using <code>gunzip</code></p>
                        <hr/>
                        <p>Go back to the main <kbd>/ncov</kbd> directory and run the basic nextstrain workflow</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">snakemake --cores 14 --profile ./my_profiles/getting_started</code></pre>
                        <hr/>
                        <p>On your own computer, not the cluster, establish a tunnel to server. Here you should use your own username</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">ssh -L 4000:localhost:4000 studXX@workshop2021a.vhost37.genap.ca</code></pre>
                        <hr/>
                        <p>Start the auspice server to view the results using your credentials</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">auspice view --datasetDir /home/studXX/projects/lec4/ncov/auspice</code></pre>
                        <hr/>
                        <p>You can visit the results page at <a href="http://localhost:4000" target="_blank">http://localhost:4000</a> in your browser. If you're done inspecting the results, terminate the auspice server with <code>Ctrl+C</code></p>
                        <hr/>
                        <p>Include our own recently assembled data. Use <kbd>consensus_sequences.fasta</kbd> and <kbd>metadata.tsv</kbd> from <code>/home/hgen_share/lec4_pt2</code>. Append the fasta file to the example sequences (you can also inspect the fasta file with <code>less</code>)</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">cat consensus_sequences.fasta >> data/example_sequences.fasta</code></pre>
                        <hr/>
                        <p>Do the same for the metadata file. Append it to <kbd>example_metadata.tsv</kbd></p>
                        <hr/>
                        <p>Run the same steps as above. Where does our data reside in the phylogeny?</p>
                    </div>
                </div>
            </div>
        </div>
        <!-- Bootstrap core JavaScript
    ================================================== -->
        <!-- Placed at the end of the document so the pages load faster -->
        <script src="assets/js/jquery.min.js"></script>
        <script src="assets/js/popper.js"></script>
        <script src="bootstrap/js/bootstrap.min.js"></script>
        <script src="assets/js/prism.js"></script>
        <script src="assets/js/extra.js"></script>
        <a id="back-to-top" href="#" class="btn btn-primary btn-lg back-to-top" role="button" data-toggle="tooltip" data-placement="left"><span class="fas fa-angle-up"></span></a>
    </body>
</html>
