<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <meta name="description" content="">
        <meta name="author" content="">
        <title>Lecture 4</title>
        <!-- Bootstrap core CSS -->
        <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
        <link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
        <link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
        <link rel="manifest" href="/site.webmanifest">
        <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5">
        <meta name="msapplication-TileColor" content="#da532c">
        <meta name="theme-color" content="#ffffff">
        <!-- Bootstrap core CSS -->
        <link href="bootstrap/css/bootstrap.css" rel="stylesheet">
        <!-- Custom styles for this template -->
        <link href="https://fonts.googleapis.com/css?family=Playfair+Display:700,900" rel="stylesheet">
        <link href="styles.css" rel="stylesheet">
        <link href="prism.css" rel="stylesheet">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Condensed">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Mono">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto+Slab">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Titillium+Web">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Dosis">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Quicksand">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto">
        <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.3.1/css/all.css">
    </head>
    <body>
        <div class="container">
            <header class="blog-header py-3 mb-4 pt-3 pb-3">
                <h1 style="font-family: Roboto Condensed; font-weight: bold;" class="display-5 mb-0 text-center">Lecture 4</h1>
            </header>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse1" aria-expanded="false" aria-controls="collapse1">Copy the files for today's class from <code>/home/partage/lec4/</code></a></h5>
                </div>
                <div id="collapse1" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Set up directory and copy over files</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">module load StdEnv/2020 gcc/9.3.0 openmpi/4.0.3 hdf5/1.10.6 minimap2/2.17 fastqc/0.11.9 samtools/1.11 mugqic/qualimap/2.2.1 quast/5.0.2 mafft/7.471 emboss/6.6.0 fasttree/2.1.11
conda activate /home/partage/envs/ont
export PATH=/home/partage/ont-guppy-cpu/bin:$PATH
mkdir -p ~/projects/lec4
cd ~/projects/lec4
cp -r /home/partage/lec4/* ./
ls -lh</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse2" aria-expanded="false" aria-controls="collapse2">Take a look at <code>ex.fast5</code></a></h5>
                </div>
                <div id="collapse2" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>View the file with <kbd>h5ls</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">h5ls ex.fast5 | less</code></pre>
                        <hr/>
                        <p>Show structure of one read using <kbd>h5ls</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">h5ls -r ex.fast5/read_00017b75-18fc-40ae-8a21-a5b3e49dc753</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse3" aria-expanded="false" aria-controls="collapse3">Basecall using Guppy</a></h5>
                </div>
                <div id="collapse3" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Show configurations available to <kbd>guppy_basecaller</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">guppy_basecaller --print_workflows</code></pre>
                        <hr/>
                        <p>Which one should you use for a FLO-MIN106 flowcell with the LSK109 kit?</p>
                        <hr/>
                        <p>Proceed to basecall. This will take a while with just 1 CPU thread, and since the output is provided already, you can terminate it prematurely with <kbd>Ctrl-C</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">guppy_basecaller \
  --compress_fastq \
  -i ex.fast5 \
  -s basecall/ \
  --cpu_threads_per_caller 1 \
  --num_callers 1 \
  -c ${conf}.cfg</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse4" aria-expanded="false" aria-controls="collapse4">Assess quality with pycoQC & FastQC</a></h5>
                </div>
                <div id="collapse4" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                      <p>Run <kbd>pycoQC</kbd> with the provided summary</p>
                      <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">pycoQC -f sequencing_summary.txt -o pyco.html</code></pre>
                      <hr/>
                        <p>Run <kbd>FastQC</kbd> with the provided sequences</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">fastqc ex.fq</code></pre>
                        <hr/>
                        <p>Move results to public folder and download from https://hgen663.vhost37.genap.ca/~studXX</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">mkdir -p ~/public_html/lec4
mv *html ~/public_html/lec4/
chmod -R 755 ~/public_html/lec4</code></pre>
                        <hr/>
                        <p>Compare with another report <a href="https://adrienleger.com/pycoQC/pycoQC/results/Guppy-2.3_basecall-1D_alignment-DNA.html" target="_blank">here</a></p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse5" aria-expanded="false" aria-controls="collapse5">Rectify quality issues by trimming</a></h5>
                </div>
                <div id="collapse5" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Trim adapters with <kbd>Porechop</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">porechop -i ex.fq -t 1 -v 2 -o chopped.fq</code></pre>
                        <hr/>
                        <p>Is it better now? https://hgen663.vhost37.genap.ca/~studXX</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">fastqc chopped.fq
mv chopped_fastqc.html ~/public_html/lec4/
chmod 644 ~/public_html/lec4/chopped_fastqc.html</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse6" aria-expanded="false" aria-controls="collapse6">Assess error profile after alignment</a></h5>
                </div>
                <div id="collapse6" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Align trimmed reads using <kbd>minimap2</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">minimap2 -t 1 -x map-ont -a wuhCor1.fa chopped.fq | samtools sort -o ex.bam
samtools index ex.bam</code></pre>
                        <hr/>
                        <p>Summarize alignment results with <kbd>Qualimap</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">qualimap bamqc -bam ex.bam --java-mem-size=1G -nw 5000 -nt 1 -c -outdir qualimap/</code></pre>
                        <hr/>
                        <p>Move results to public folder and download from https://hgen663.vhost37.genap.ca/~studXX</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">mv qualimap ~/public_html/lec4/
chmod -R 755 ~/public_html/lec4/qualimap</code></pre>
                        <hr/>
                        <p>Compare with the <a href="https://rawgit.com/kokonech/kokonech.github.io/master/qualimap/HG00096.chr20_bamqc/qualimapReport.html" target="_blank">report</a> from a typical Illumina run</p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse7" aria-expanded="false" aria-controls="collapse7">(Optional) Hybrid assembly with Unicycler</a></h5>
                </div>
                <div id="collapse7" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                      <p>Copy files over</p>
                      <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">cp /home/partage/lec4_extra/*fq ./</code></pre>
                      <hr/>
                      <p>Run <kbd>Unicycler</kbd> using boths short- and long-reads</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">unicycler -1 Illumina_F.fq -2 Illumina_R.fq -l ONT.fq -t 20 --min_fasta_length 10000 --linear_seqs 1 -o unicycler</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse8" aria-expanded="false" aria-controls="collapse8">Evaluate the assembled genome's quality</a></h5>
                </div>
                <div id="collapse8" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Visualize contigs with <kbd>Bandage</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">Bandage image unicycler/assembly.gfa assembly.jpg</code></pre>
                        <hr/>
                        <p>Use <kbd>QUAST</kbd> to create a summary report</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">quast.py -t 1 -o quast -R wuhCor1.fa -g wuhCor1.gff3 unicycler/assembly.fasta</code></pre>
                        <hr/>
                        <p>Move results to public folder and download from https://hgen663.vhost37.genap.ca/~studXX</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">mv -t ~/public_html/lec4/ quast assembly.jpg
chmod -R 755 ~/public_html/lec4/</code></pre>
                        <hr/>
                        <p>Compare with a <a href="http://cab.cc.spbu.ru/quast/paper/h.sapiens_chr14/" target="_blank">report</a> on eukaryotic assemblies</p>
                        <hr/>
                        <p><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="_blank">BLAST</a> the top 2 contigs (<code>contig1.fa</code> and <code>contig2.fa</code>)</p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse9" aria-expanded="false" aria-controls="collapse9">Align the CDS for S proteins from various species in <code>s.fna</code></a></h5>
                </div>
                <div id="collapse9" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Translate nucleotides sequences into those of amino acids with <kbd>transeq</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">transeq s.fna s.faa</code></pre>
                        <hr/>
                        <p>Align protein sequences using <kbd>mafft</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">mafft --auto s.faa > s.align.faa</code></pre>
                        <hr/>
                        <p>Use aligned protein sequences to generate aligned nucleic coding regions via <kbd>tranalign</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">tranalign s.fna s.align.faa s.align.fna</code></pre>
                        <hr/>
                        <p>Browser results from NCBI's <a href="https://www.ncbi.nlm.nih.gov/projects/msaviewer/" target="_blank">Multiple Sequence Alignment Viewer</a></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">mv s.align.fna ~/public_html/lec4/
chmod 644 ~/public_html/lec4/s.align.fna</code></pre>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse10" aria-expanded="false" aria-controls="collapse10">Use HyPhy selection using a subset of S CDSs listed in <code>s.sub.fna</code></a></h5>
                </div>
                <div id="collapse10" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Align cDNA sequences like before</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">transeq s.sub.fna s.sub.faa
mafft --auto s.sub.faa > s.sub.align.faa
tranalign s.sub.fna s.sub.align.faa s.sub.align.fna</code></pre>
                        <hr/>
                        <p>Infer phylogeny with <kbd>FastTree</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">FastTree -gtr -nt s.sub.align.fna > s.sub.align.nhx</code></pre>
                        <hr/>
                        <p>Explore the presence of diversifying selection using <kbd>aBSREL</kbd></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">hyphy absrel --alignment s.sub.align.fna --tree s.sub.align.nhx</code></pre>
                        <hr/>
                        <p>(Optional) Identify recombination breakpoints with GARD</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">hyphy gard s.sub.align.fna</code></pre>
                        <hr/>
                        <p>Visualize results (<code>*.json</code> files) using <a href="http://vision.hyphy.org/" target="_blank">http://vision.hyphy.org/</a></p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">mv *json ~/public_html/lec4/
chmod 644 ~/public_html/lec4/*json</code></pre>
                        <hr/>
                        <p>Look at the <a href="https://observablehq.com/@spond/revised-sars-cov-2-analytics-page" target="_blank">results</a> of selection analysis enabled by a much larger dataset</p>
                    </div>
                </div>
            </div>
            <div class="card mb-4 hoverable">
                <div class="card-header" role="tab">
                    <h5 class="mb-0"><a class="collapsed" data-toggle="collapse" href="#collapse11" aria-expanded="false" aria-controls="collapse11">(Optional) Place newly assembled viral genomes in a global context</a></h5>
                </div>
                <div id="collapse11" class="collapse" role="tabpanel" aria-labelledby="headingThree" data-parent="">
                    <div class="card-body">
                        <p>Activate the nextstrain environment</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">conda deactivate
conda activate /home/partage/envs/nextstrain
export LD_LIBRARY_PATH=/home/partage/envs/nextstrain/lib:$LD_LIBRARY_PATH</code></pre>
                        <hr/>
                        <p>Set up directory with example data</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">cd ~/projects/lec4/
git clone https://github.com/nextstrain/ncov.git
gunzip ncov/data/example_sequences.fasta.gz</code></pre>
                        <hr/>
                        <p>Add pre-computed assemblies along with their metadata, which you could take a peek at as well</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">cat /home/partage/lec4_extra/consensus_sequences.fasta >> ncov/data/example_sequences.fasta
cat /home/partage/lec4_extra/metadata.tsv >> ncov/data/example_metadata.tsv</code></pre>
                        <hr/>
                        <p>Run pipeline</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">cd ncov
snakemake --cores 20 --profile ./my_profiles/getting_started</code></pre>
                        <hr/>
                        <p>Start visualization server</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">nextstrain view --port 4000 auspice/</code></pre>
                        <hr/>
                        <p>On your own computer, not the cluster, establish a tunnel to server. Here you should use your own username</p>
                        <pre class="language-bash" data-user="user" data-host="remotehost"><code class="language-bash">ssh -L 8989:localhost:4000 studXX@hgen663.vhost37.genap.ca</code></pre>
                        <hr/>
                        <p>Now you can visit it in your browser at <a href="http://localhost:8989" target="_blank">http://localhost:8989</a>. If port 4000 is not available (i.e., another student is using it), then you can change both mentions of it to something else (e.g., 4001). Similarly, if port 8989 on your own computer is unavailable, you can also change that</p>
                    </div>
                </div>
            </div>
        </div>
        <!-- Bootstrap core JavaScript
    ================================================== -->
        <!-- Placed at the end of the document so the pages load faster -->
        <script src="assets/js/jquery.min.js"></script>
        <script src="assets/js/popper.js"></script>
        <script src="bootstrap/js/bootstrap.min.js"></script>
        <script src="assets/js/prism.js"></script>
        <script src="assets/js/extra.js"></script>
        <a id="back-to-top" href="#" class="btn btn-primary btn-lg back-to-top" role="button" data-toggle="tooltip" data-placement="left"><span class="fas fa-angle-up"></span></a>
    </body>
</html>
